src/api/firebase.js

```diff
@@ -173,3 +173,71 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+/**
+ * Compare purchase history of list items and sort items by three attributes:
+ * 1) Purchase status, 2) Urgency, and 3) Alphabetically
+ * add a property to each item indicating the urgency of the next purchase
+ * @param {object} data The raw list data to be sorted.
+ */
+export function comparePurchaseUrgency(data) {
+	// Create an empty, nested array for sorted data categories
+	const sortedData = [];
+	for (let i = 0; i < 6; i++) {
+		sortedData.push([]);
+	}
+
+	// Loop through all items and assign urgency based on days until next purchase
+	for (let item of data) {
+		// If the item has been purchased before, return days since last purchase
+		// If the item has never been purchased, return the days since item creation
+		const daysSincePurchase = item.dateLastPurchased
+			? getDaysBetweenDates(item.dateLastPurchased)
+			: getDaysBetweenDates(item.dateCreated);
+
+		// Assign urgency
+		if (daysSincePurchase >= 60) {
+			item.purchaseStatus = 'Inactive';
+			item.urgencyCategory = 4; // Inactive
```

Comment:
Directly updating the input data (here you're adding extra properties) without creating a clone can lead to some odd/unintended behavior in your business logic.

I would suggest either
1. cloning the item
2. refactor to omit the use of `purchaseStatus` and `urgencyCategory`

I recommend number 2. (More on that below)

Reply:
I'll take door # 2!